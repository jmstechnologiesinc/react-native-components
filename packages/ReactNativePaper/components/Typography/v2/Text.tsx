import * as React from 'react';
import {
  Text as NativeText,
  TextStyle,
  StyleProp,
  StyleSheet,
} from 'react-native';

import theme from '../../../styles/themes/v3/LightTheme';

type Props = React.ComponentProps<typeof NativeText> & {
  style?: StyleProp<TextStyle>;
  /**
   * @optional
   */
};

// @component-group Typography

/**
 * Text component which follows styles from the theme.
 *
 * @extends Text props https://reactnative.dev/docs/text#props
 */
const Text: React.ForwardRefRenderFunction<{}, Props> = (
  {style, ...rest}: Props,
  ref,
) => {
  const root = React.useRef<NativeText | null>(null);

  React.useImperativeHandle(ref, () => ({
    setNativeProps: (args: Object) => root.current?.setNativeProps(args),
  }));

  return (
    <NativeText
      {...rest}
      ref={root}
      style={[
        {
          ...(!theme.isV3 && theme.fonts?.regular),
          color: theme.isV3 ? theme.colors?.onSurface : theme.colors.text,
        },
        styles.text,
        style,
      ]}
    />
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'left',
  },
});

export default React.forwardRef(Text);
