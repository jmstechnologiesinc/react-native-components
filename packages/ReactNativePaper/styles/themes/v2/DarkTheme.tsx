import color from 'color';
import LightTheme from './LightTheme';
import { black, white, pinkA100 } from './colors';
import configureFonts from '../../fonts';
import type { ThemeBase } from '../../../types';

const DarkTheme: ThemeBase = {
  ...LightTheme,
  dark: true,
  mode: 'adaptive',
  version: 2,
  isV3: false,
  colors: {
    ...LightTheme.colors,
    primary: '#BB86FC',
    accent: '#03dac6',
    background: '#121212',
    surface: '#121212',
    error: '#CF6679',
    onSurface: '#FFFFFF',
    text: white,
    disabled: color(white).alpha(0.38).rgb().string(),
    placeholder: color(white).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    notification: pinkA100,
  },
  fonts: configureFonts(),
};

export default DarkTheme;
