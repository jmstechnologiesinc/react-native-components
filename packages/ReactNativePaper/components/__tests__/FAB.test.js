import * as React from 'react';
import renderer from 'react-test-renderer';
import color from 'color';
import FAB from '../FAB';
import { getFABColors } from '../FAB/utils';
import getContrastingColor from '../../utils/getContrastingColor';
import { black, white } from '../../styles/themes/v2/colors';

import MD3LightTheme from '../../styles/themes/v3/LightTheme';
import MD2LightTheme from '../../styles/themes/v2/LightTheme';
import MD3DarkTheme from '../../styles/themes/v3/DarkTheme';
import MD2DarkTheme from '../../styles/themes/v2/DarkTheme';

const getTheme = (isDark = false, isV3 = true) => {
  const theme = isDark
    ? isV3
      ? MD3DarkTheme
      : MD2DarkTheme
    : isV3
    ? MD3LightTheme
    : MD2LightTheme;
  return {
    ...theme,
    isV3,
  };
};

it('renders normal FAB', () => {
  const tree = renderer.create(<FAB onPress={() => {}} icon="plus" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders small FAB', () => {
  const tree = renderer
    .create(<FAB size="small" onPress={() => {}} icon="plus" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders extended FAB', () => {
  const tree = renderer
    .create(<FAB onPress={() => {}} icon="plus" label="Add items" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders loading FAB', () => {
  const tree = renderer
    .create(<FAB onPress={() => {}} icon="plus" loading={true} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders disabled FAB', () => {
  const tree = renderer
    .create(<FAB onPress={() => {}} icon="plus" disabled />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders custom color for the icon and label of the FAB', () => {
  const tree = renderer
    .create(<FAB onPress={() => {}} icon="plus" color="#AA0114" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders not visible FAB', () => {
  const { update, toJSON } = renderer.create(
    <FAB onPress={() => {}} icon="plus" />
  );
  update(<FAB onPress={() => {}} icon="plus" visible={false} />);

  expect(toJSON()).toMatchSnapshot();
});

it('renders visible FAB', () => {
  const { update, toJSON } = renderer.create(
    <FAB onPress={() => {}} icon="plus" visible={false} />
  );

  update(<FAB onPress={() => {}} icon="plus" visible={true} />);

  expect(toJSON()).toMatchSnapshot();
});

describe('getFABColors - background color', () => {
  it('should return color from styles', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'primary',
        style: { backgroundColor: 'purple' },
      })
    ).toMatchObject({
      backgroundColor: 'purple',
    });
  });

  it('should return correct disabled color, for theme version 3', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        disabled: true,
        variant: 'primary',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.surfaceDisabled,
    });
  });

  it('should return correct disabled color, for theme version 2, light mode', () => {
    expect(
      getFABColors({
        theme: getTheme(false, false),
        disabled: true,
        variant: 'primary',
      })
    ).toMatchObject({
      backgroundColor: color(black).alpha(0.12).rgb().string(),
    });
  });

  it('should return correct disabled color, for theme version 2, dark mode', () => {
    expect(
      getFABColors({
        theme: getTheme(true, false),
        disabled: true,
        variant: 'primary',
      })
    ).toMatchObject({
      backgroundColor: color(white).alpha(0.12).rgb().string(),
    });
  });

  it('should return correct theme color, for theme version 3, primary variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'primary',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.primaryContainer,
    });
  });

  it('should return correct theme color, for theme version 3, secondary variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'secondary',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.secondaryContainer,
    });
  });

  it('should return correct theme color, for theme version 3, tertiary variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'tertiary',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.tertiaryContainer,
    });
  });

  it('should return correct theme color, for theme version 3, surface variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'surface',
      })
    ).toMatchObject({
      backgroundColor: getTheme().colors.elevation.level3,
    });
  });

  it('should return correct theme color, for theme version 2', () => {
    expect(
      getFABColors({
        theme: getTheme(false, false),
        variant: 'primary',
      })
    ).toMatchObject({
      backgroundColor: getTheme(false, false).colors.accent,
    });
  });
});

describe('getFABColors - foreground color', () => {
  it('should return custom color', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'primary',
        customColor: 'purple',
      })
    ).toMatchObject({
      foregroundColor: 'purple',
    });
  });

  it('should return correct disabled color, for theme version 3', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'primary',
        disabled: true,
      })
    ).toMatchObject({
      foregroundColor: getTheme().colors.onSurfaceDisabled,
    });
  });

  it('should return correct disabled color, for theme version 2, light mode', () => {
    expect(
      getFABColors({
        theme: getTheme(false, false),
        disabled: true,
        variant: 'primary',
      })
    ).toMatchObject({
      foregroundColor: color(black).alpha(0.32).rgb().string(),
    });
  });

  it('should return correct disabled color, for theme version 2, dark mode', () => {
    expect(
      getFABColors({
        theme: getTheme(true, false),
        disabled: true,
        variant: 'primary',
      })
    ).toMatchObject({
      foregroundColor: color(white).alpha(0.32).rgb().string(),
    });
  });

  it('should return correct theme color, for theme version 3, primary variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'primary',
      })
    ).toMatchObject({
      foregroundColor: getTheme().colors.onPrimaryContainer,
    });
  });

  it('should return correct theme color, for theme version 3, secondary variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'secondary',
      })
    ).toMatchObject({
      foregroundColor: getTheme().colors.onSecondaryContainer,
    });
  });

  it('should return correct theme color, for theme version 3, tertiary variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'tertiary',
      })
    ).toMatchObject({
      foregroundColor: getTheme().colors.onTertiaryContainer,
    });
  });

  it('should return correct theme color, for theme version 3, surface variant', () => {
    expect(
      getFABColors({
        theme: getTheme(),
        variant: 'surface',
      })
    ).toMatchObject({
      foregroundColor: getTheme().colors.primary,
    });
  });

  it('should return correct theme color, for theme version 2', () => {
    expect(
      getFABColors({
        theme: getTheme(false, false),
        variant: 'primary',
      })
    ).toMatchObject({
      foregroundColor: getContrastingColor(
        getTheme(false, false).colors.accent,
        white,
        'rgba(0, 0, 0, .54)'
      ),
    });
  });
});
