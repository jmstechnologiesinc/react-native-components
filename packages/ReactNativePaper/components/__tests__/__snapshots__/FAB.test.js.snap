// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom color for the icon and label of the FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "#AA0114",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders disabled FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(28, 27, 31, 0.12)",
          "borderRadius": 16,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 0.38)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders extended FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityLabel="Add items"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "paddingHorizontal": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(33, 0, 93, 1)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
          <Text
            selectable={false}
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "marginHorizontal": 8,
                  },
                  false,
                  Object {
                    "color": "rgba(33, 0, 93, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                ],
              ]
            }
          >
            Add items
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders loading FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            accessibilityRole="progressbar"
            accessibilityState={
              Object {
                "busy": true,
              }
            }
            accessible={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 18,
                  "opacity": 1,
                  "width": 18,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 18,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 18,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 9,
                        "overflow": "hidden",
                        "width": 18,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 18,
                          "transform": Array [
                            Object {
                              "translateY": 0,
                            },
                            Object {
                              "rotate": "-165deg",
                            },
                          ],
                          "width": 18,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 9,
                            "overflow": "hidden",
                            "width": 18,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "borderColor": "rgba(33, 0, 93, 1)",
                              "borderRadius": 9,
                              "borderWidth": 1.8,
                              "height": 18,
                              "width": 18,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 18,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 18,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 9,
                        "overflow": "hidden",
                        "top": 9,
                        "width": 18,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 18,
                          "transform": Array [
                            Object {
                              "translateY": -9,
                            },
                            Object {
                              "rotate": "345deg",
                            },
                          ],
                          "width": 18,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 9,
                            "overflow": "hidden",
                            "width": 18,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "borderColor": "rgba(33, 0, 93, 1)",
                              "borderRadius": 9,
                              "borderWidth": 1.8,
                              "height": 18,
                              "width": 18,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders normal FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(33, 0, 93, 1)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders not visible FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(33, 0, 93, 1)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders small FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 12,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 12,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 12,
                "height": 40,
                "width": 40,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(33, 0, 93, 1)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders visible FAB 1`] = `
<View
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderRadius": 16,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 16,
                "height": 56,
                "width": 56,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "rgba(33, 0, 93, 1)",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
